libs/designer/src/lib/core/queries/connector.ts

```diff
@@ -219,6 +219,7 @@ export const getLegacyDynamicTreeItems = async (
     return {
       value,
       displayName: (getPropertyValue(value, titlePath as string) ?? '').toString(),
+      id: value.Id,
```

Comment:
Currently `id` is set up to be optional, so it should be appropriately handled downstream if not present. However, you raise a good point here, and it would prevent issues with duplicate keys if multiple have undefined IDs. I will update that.

Reply:
Fixed in 25fab9bd63cae28edeab1fdd7f3e03e6eb2b930b.